src/optimizer/plugins/AddPathRulesOptimizer.ts

```diff
@@ -0,0 +1,61 @@
+import { Package } from '../../processor';
+import { OptimizerPlugin } from '../OptimizerPlugin';
+import SimplifyArrayIndexingOptimizer from './SimplifyArrayIndexingOptimizer';
+import { ProcessingOptions } from '../../utils';
+import { ExportablePathRule, ExportableSdRule } from '../../exportable';
+import { cloneDeep } from 'lodash';
+import SimplifyRulePathContextsOptimizer from './SimplifyRulePathContextsOptimizer';
+
+export default {
+  name: 'add_path_rules',
+  description: 'Add path rules to support indenting.',
+  runBefore: [SimplifyArrayIndexingOptimizer.name, SimplifyRulePathContextsOptimizer.name],
+  optimize(pkg: Package): void {
+    [
+      ...pkg.profiles,
+      ...pkg.extensions,
+      ...pkg.logicals,
+      ...pkg.resources,
+      ...pkg.instances
```

Comment:
I think Mappings can have Path Rules on them as well, right? Should those be included here as well?

Reply:
I didn't think so, but you are exactly right:
```
mappingEntityRule:  mappingRule | insertRule | pathRule;
```
So yes, we should add that here I guess.  I also didn't realize we supported insert rules on mappings.  Go figure.

Reply:
This is another fact I re-learned when I was also re-learning that SUSHI doesn't have a true PathRule class. Oops.

Reply:
Added `mappings` in e0dd13a.